### PR TITLE
Fix incorrect calculation in `indexTree.treePosToPath` operation

### PR DESCRIPTION
**Pull request by raararaara**
**What this PR does / why we need it**:

This patch addresses an issue arising from the `tree.edit` operation in response to problem with calculating the `fromPath` when multiple `treePos` represent a single index-based coordinate, leading to inaccuracies.

This PR contains migration work for https://github.com/yorkie-team/yorkie-js-sdk/pull/751, so detailed information can also be found there.

**Which issue(s) this PR fixes**:

Address [#749](https://github.com/yorkie-team/yorkie-js-sdk/issues/749)
Related https://github.com/yorkie-team/yorkie-js-sdk/pull/751

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: pkg/document/crdt/tree.go [hunk 1]
```diff
@@ -1030,6 +1030,19 @@ func (t *Tree) ToIndex(parentNode, leftSiblingNode *TreeNode) (int, error) {
 	return idx, nil
 }
 
+// ToPath returns path from given CRDTTreePos
+func (t *Tree) ToPath(parentNode, leftSiblingNode *TreeNode) ([]int, error) {
+	treePos, err := t.toTreePos(parentNode, leftSiblingNode)
+	if err != nil {
+		return make([]int, 0), err
```
Comment (hackerwins) [inline pkg/document/crdt/tree.go:1037]:
```suggestion
		return nil, err
```

https://github.com/uber-go/guide/blob/master/style.md#nil-is-a-valid-slice

File: pkg/index/tree.go [hunk 2]
```diff
@@ -593,12 +593,16 @@ func (n *Node[V]) InsertAfter(newNode, referenceNode *Node[V]) error {
 	return nil
 }
 
-// HasTextChild returns true if the node has a text child.
+// HasTextChild returns true if the node's children consist of only text children.
 func (n *Node[V]) HasTextChild() bool {
-	for _, child := range n.Children() {
-		if child.IsText() {
-			return true
+	if len(n.children) > 0 {
```
Comment (hackerwins) [inline pkg/index/tree.go:598]: `n.children` does not filter out tombstone nodes. Is this the correct conditional expression?

File: test/integration/tree_test.go [hunk 3]
```diff
@@ -375,6 +375,72 @@ func TestTree(t *testing.T) {
...
+		fromPath, err := d2.Root().GetTree("t").ToPath(fromParent, fromLeft)
+		assert.NoError(t, err)
+		assert.Equal(t, []int{0, 5}, fromPath)
+
+		assert.NoError(t, c2.Sync(ctx))
+		assert.NoError(t, c1.Sync(ctx))
+
+		assert.Equal(t, "<r><p>hel3mo</p></r>", d1.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><p>hel3mo</p></r>", d2.Root().GetTree("t").ToXML())
+	})
```
Comment (hackerwins) [inline test/integration/tree_test.go:442]: It would be nice if the test code was grouped by step with new lines. Could you explain the steps in this test?

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for your contribution.
I left simple comments.

File: pkg/document/crdt/tree.go (hunk 1 above)
Comment (raararaara) [reply pkg/document/crdt/tree.go:1037]: Thanks for suggestion, I will reflect the corrections according to the guide you mentioned.

File: pkg/index/tree.go (hunk 2 above)
Comment (raararaara) [reply pkg/index/tree.go:598]: No, it wasn't intended, and it's not a correct conditional expression. I will fix it.

File: test/integration/tree_test.go (hunk 3 above)
Comment (raararaara) [reply test/integration/tree_test.go:442]: I will add a description of each step as a comment.

**Review by hackerwins — APPROVED** [review]
LGTM.

**Pull request merged**